### ISSUE TITLE
Upon destroy of eks component "depends on resource attributes that cannot be determined"

### ISSUE THREAD

**Issue by nitrocode** [cloudposse/terraform-yaml-stack-config#54]
Found a bug? Maybe our [Slack Community](https://slack.cloudposse.com) can help. 

[![Slack Community](https://slack.cloudposse.com/badge.svg)](https://slack.cloudposse.com)

## Describe the Bug
When I destroy the eks component, I see this error

```hcl
│ Error: Invalid count argument
│
│   on .terraform/modules/iam_roles.account_map/modules/remote-state/remote.tf line 6, in data "terraform_remote_state" "remote":
│    6:   count = local.remote_state_enabled && local.backend_type == "remote" ? 1 : 0
│
│ The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this, use the -target
│ argument to first apply only the resources that the count depends on.
╵
╷
│ Error: Invalid count argument
│
│   on .terraform/modules/iam_roles.account_map/modules/remote-state/s3.tf line 6, in data "terraform_remote_state" "s3":
│    6:   count = local.remote_state_enabled && local.backend_type == "s3" ? 1 : 0
│
│ The "count" value depends on resource attributes that cannot be determined until apply, so Terraform cannot predict how many instances will be created. To work around this, use the -target
│ argument to first apply only the resources that the count depends on.
```

## Expected Behavior
Allow destruction

## Steps to Reproduce
1. `atmos terraform destroy eks --stack uw2-dev`

## Additional Context

https://github.com/cloudposse/terraform-yaml-stack-config/blob/18239bde1692f405a121a5d1b2daf8e2bfd0dace/modules/remote-state/remote.tf#L5-L6

https://github.com/cloudposse/terraform-yaml-stack-config/blob/18239bde1692f405a121a5d1b2daf8e2bfd0dace/modules/remote-state/s3.tf#L5-L6

https://github.com/cloudposse/terraform-yaml-stack-config/blob/18239bde1692f405a121a5d1b2daf8e2bfd0dace/modules/remote-state/variables.tf#L30-L34

https://github.com/cloudposse/terraform-yaml-stack-config/blob/18239bde1692f405a121a5d1b2daf8e2bfd0dace/modules/remote-state/main.tf#L34

https://github.com/cloudposse/terraform-yaml-stack-config/blob/18239bde1692f405a121a5d1b2daf8e2bfd0dace/modules/remote-state/main.tf#L18

## workarounds

I tried setting `bypass = ! var.enabled` which resulted in the same above error.

The workaround was to destroy the eks component using `enabled=false` which got around this error.

**Comment by Nuru** [comment]
Closing as duplicate of https://github.com/hashicorp/terraform/issues/27849 which was [resolved in Terraform 1.0](https://github.com/hashicorp/terraform/pull/28017). Note that even using Terraform 1.3, this issue will arise if the last `terraform apply` was from a broken version.

**Comment by Nuru** [comment]
Looks like this is also a new bug: https://github.com/hashicorp/terraform/issues/32023

**Comment by Nuru** [comment]
Closing as "fixed" in #56 because there is not much else we can do until https://github.com/hashicorp/terraform/issues/32023 is fixed.
